compiler/src/dmd/expression.d

```diff
@@ -566,6 +565,8 @@ extern (C++) final class IntegerExp : Expression
         /* 'Normalize' the value of the integer to be in range of the type
          */
         dinteger_t result;
+        if (ty == Tpointer)
+            ty = Type.tsize_t.ty;
```

Comment:
Strictly, the size of pointers and size_t are unconnected. There is hardware or systems with bigger pointer sizes than the available addressable memory.

Reply:
Technically, x86-64 is one of those systems. The segment registers are still used for thread-local variables, so we'd have a >64-bit far pointer there. Thankfully, we don't actually have to care about that in practice.

The 65c816, which I've experimented with with D, has 16-bit pointers and a 24-bit address space. 24-bit pointers are possible, but significantly more expensive (and currently unsupported by the compiler backend) due to the 16-bit registers. I would LIKE D to care about that, but I'd be pretty surprised to see it...

Either way, it'd be better to say what we mean instead of relying on things that are _usually_ true.